### ISSUE TITLE
Sudden loss spike at ~0.4 epoch during training on 4×A100 — expected behavior?

### ISSUE THREAD

**Issue by xXuHaiyang** [ximinng/LLM4SVG#19]
Hi, 

Thank you for the great work! 

While training the model following the exact setup in the README (4×A100), I observed a sudden spike in loss around epoch 0.4 — it jumped from ~0.45 to over 3.0 in one step, and the grad_norm also increased drastically at that point. See the training logs below for reference:

![Image](https://github.com/user-attachments/assets/d2bb4ea0-3679-4d95-ad2c-b1f89e092d26)

This looks like a sharp outlier compared to the otherwise smooth training curve before and after.

Questions:
- Is this behavior expected or known during early training?
- Could this be related to learning rate scheduling, gradient accumulation, or optimizer dynamics?
- Any recommendations for mitigation or whether it should be a concern?

Thanks in advance!

**Comment by ericaweng** [comment]
which model / config are you using?

**Comment by IApple233** [comment]
Hi, I also met the same problem. Have you solved it?

**Comment by IApple233** [comment]
I changed the dataset, and I found that, no matter what dataset or combination of datasets you use, there is all a  loss spike  at around 0.4 epoch.

**Comment by IApple233** [comment]
I use llama-factory to fine-tune Qwen-VL 7B

**Comment by Qingman-Wu** [comment]
> I use llama-factory to fine-tune Qwen-VL 7B

same question

Have you solved it?

**Comment by IApple233** [comment]
> > I use llama-factory to fine-tune Qwen-VL 7B
> 
> same question
> 
> Have you solved it?

In my experiment. when I do fine-tune on OmniSVG's base model, although there is still a strange spike, but after the spike, the loss curve can quickly become normal and decrease to 0.4 and lower

**Comment by leesabak** [comment]
> > > I use llama-factory to fine-tune Qwen-VL 7B
> > 
> > 
> > same question
> > Have you solved it?
> 
> In my experiment. when I do fine-tune on OmniSVG's base model, although there is still a strange spike, but after the spike, the loss curve can quickly become normal and decrease to 0.4 and lower

May I ask if you have tried framework fine-tuning for OmniSVG-3b, such as Llama factory

**Comment by IApple233** [comment]
> > > > I use llama-factory to fine-tune Qwen-VL 7B
> > > 
> > > 
> > > same question
> > > Have you solved it?
> > 
> > 
> > In my experiment. when I do fine-tune on OmniSVG's base model, although there is still a strange spike, but after the spike, the loss curve can quickly become normal and decrease to 0.4 and lower
> 
> May I ask if you have tried framework fine-tuning for OmniSVG-3b, such as Llama factory

yeah, I have tried it.

**Comment by leesabak** [comment]
> > > > > I use llama-factory to fine-tune Qwen-VL 7B
> > > > 
> > > > 
> > > > same question
> > > > Have you solved it?
> > > 
> > > 
> > > In my experiment. when I do fine-tune on OmniSVG's base model, although there is still a strange spike, but after the spike, the loss curve can quickly become normal and decrease to 0.4 and lower
> > 
> > 
> > May I ask if you have tried framework fine-tuning for OmniSVG-3b, such as Llama factory
> 
> yeah, I have tried it.

Now I'm attempting to fine-tuning omnisvg on my own dataset, but cant find any info about how to tokenize those tokens.
I wonder do u have an plan to release the file or do I have other way to access it? Thanks so much!

**Comment by IApple233** [comment]
> > > > > > I use llama-factory to fine-tune Qwen-VL 7B
> > > > > 
> > > > > 
> > > > > same question
> > > > > Have you solved it?
> > > > 
> > > > 
> > > > In my experiment. when I do fine-tune on OmniSVG's base model, although there is still a strange spike, but after the spike, the loss curve can quickly become normal and decrease to 0.4 and lower
> > > 
> > > 
> > > May I ask if you have tried framework fine-tuning for OmniSVG-3b, such as Llama factory
> > 
> > 
> > yeah, I have tried it.
> 
> Now I'm attempting to fine-tuning omnisvg on my own dataset, but cant find any info about how to tokenize those tokens. I wonder do u have an plan to release the file or do I have other way to access it? Thanks so much!

I think you should refer to the token-to-SVG process in Omnisvg and rewrite a new tokenizer yourself. I will release it once I have completed my project.

**Comment by leesabak** [comment]
> > > > > > > I use llama-factory to fine-tune Qwen-VL 7B
> > > > > > 
> > > > > > 
> > > > > > same question
> > > > > > Have you solved it?
> > > > > 
> > > > > 
> > > > > In my experiment. when I do fine-tune on OmniSVG's base model, although there is still a strange spike, but after the spike, the loss curve can quickly become normal and decrease to 0.4 and lower
> > > > 
> > > > 
> > > > May I ask if you have tried framework fine-tuning for OmniSVG-3b, such as Llama factory
> > > 
> > > 
> > > yeah, I have tried it.
> > 
> > 
> > Now I'm attempting to fine-tuning omnisvg on my own dataset, but cant find any info about how to tokenize those tokens. I wonder do u have an plan to release the file or do I have other way to access it? Thanks so much!
> 
> I think you should refer to the token-to-SVG process in Omnisvg and rewrite a new tokenizer yourself. I will release it once I have completed my project.

oh! are you have rewritten a new tokenizer by yourself? is it efficient?

**Comment by leesabak** [comment]
> > > > > > > I use llama-factory to fine-tune Qwen-VL 7B
> > > > > > 
> > > > > > 
> > > > > > same question
> > > > > > Have you solved it?
> > > > > 
> > > > > 
> > > > > In my experiment. when I do fine-tune on OmniSVG's base model, although there is still a strange spike, but after the spike, the loss curve can quickly become normal and decrease to 0.4 and lower
> > > > 
> > > > 
> > > > May I ask if you have tried framework fine-tuning for OmniSVG-3b, such as Llama factory
> > > 
> > > 
> > > yeah, I have tried it.
> > 
> > 
> > Now I'm attempting to fine-tuning omnisvg on my own dataset, but cant find any info about how to tokenize those tokens. I wonder do u have an plan to release the file or do I have other way to access it? Thanks so much!
> 
> I think you should refer to the token-to-SVG process in Omnisvg and rewrite a new tokenizer yourself. I will release it once I have completed my project.



> > > > > > > > I use llama-factory to fine-tune Qwen-VL 7B
> > > > > > > 
> > > > > > > 
> > > > > > > same question
> > > > > > > Have you solved it?
> > > > > > 
> > > > > > 
> > > > > > In my experiment. when I do fine-tune on OmniSVG's base model, although there is still a strange spike, but after the spike, the loss curve can quickly become normal and decrease to 0.4 and lower
> > > > > 
> > > > > 
> > > > > May I ask if you have tried framework fine-tuning for OmniSVG-3b, such as Llama factory
> > > > 
> > > > 
> > > > yeah, I have tried it.
> > > 
> > > 
> > > Now I'm attempting to fine-tuning omnisvg on my own dataset, but cant find any info about how to tokenize those tokens. I wonder do u have an plan to release the file or do I have other way to access it? Thanks so much!
> > 
> > 
> > I think you should refer to the token-to-SVG process in Omnisvg and rewrite a new tokenizer yourself. I will release it once I have completed my project.
> 
> oh! are you have rewritten a new tokenizer by yourself? is it efficient?

this is my email 1902405366@qq.com ，can u give me a Contact Information
